doc/mgmt/OpenConfig_System.md

```diff
@@ -0,0 +1,385 @@
+We are updating openconfig-system yang version (0.7.0) in sonic with latest available openconfig version (revision 2.1.0), to have all the latest changes from community.  
+Community PR [https://github.com/sonic-net/sonic-mgmt-common/pull/147]  
+   * In this PR openconfig-system.yang and its submodules are updated to latest version available
+   * Reference to openconfig-network-instance model is commented, as this module is yet to be supported in SONiC
+
+Main changes in the latest openconfig versions are:
+ * Feature wise major changes -
+    * system/state -> up-time, software-version, last-configuration-timestamp nodes are added.
+    * ntp -> source-address moved to per server list from global container. Also network-instance is included per server. NTP Key reference is added in server.
+    * grpc-server -> Restructured completely, multiple server provision is added.
+    * logging -> For remote-server, network-instance support is added. Files & VTY containers are added newly.
+    * memory -> used and free leaves are added.
+    * process -> uptime node is removed.
+
+ * New features added in latest version -
+    * license
+    * mac-address
+    * hashing
+    * bootz
+    * control-plane-traffic (copp)
+    * resource utilization
+
+### 3.3.2 REST API Support
+#### 3.3.2.1 GET
+Supported
+
+#### 3.3.2.2 SET
+Supported
+
+#### 3.3.2.3 DELETE
+Supported
+
+### 3.3.3 gNMI Support
+#### 3.3.3.1 GET
+Supported
+
+Sample GET output for system/state container:
+```
+gnmic -a <ip:port> -u <user> -p <passwd> get --path "/openconfig-system:system"
+[
+ {
+  "source": "<ip:port>",
+  "timestamp": 1712673160649745928,
+  "time": "2024-04-09T20:02:40.649745928+05:30",
+  "updates": [
+   {
+    "Path": "openconfig-system:system",
+    "values": {
+      "openconfig-system:system": {
+          "state": {
+            "boot-time": "1712672712",
+            "current-datetime": "2024-04-09T14:32:40Z+00:00"
+          }
+        }
+    }
+   }
+  ]
+ }
+]
+
+```
+
+#### 3.3.3.2 SET
+Supported
+Sample SET logs for system/config/hostname node:
+```
+gnmi_set -target_addr <ip:port> -update /openconfig-system:system/config/hostname:@./hostname-value.json -xpath_target OC-YANG
+/openconfig-system:system/config/hostname
+@./hostname-value.json
+== setRequest:
+prefix: <
+  target: "OC-YANG"
+>
+update: <
+  path: <
+    origin: "openconfig-system"
+    elem: <
+      name: "system"
+    >
+    elem: <
+      name: "config"
+    >
+    elem: <
+      name: "hostname"
+    >
+  >
+  val: <
+    json_ietf_val: "{\"hostname\": \"mytest-hostname\"}"
+  >
+>
+
+== setResponse:
+prefix: <
+  target: "OC-YANG"
+>
+response: <
+  path: <
+    origin: "openconfig-system"
+    elem: <
+      name: "openconfig-system:system"
+    >
+    elem: <
+      name: "config"
+    >
+    elem: <
+      name: "hostname"
+    >
+  >
+  op: UPDATE
+>
+```
+
+#### 3.3.3.3 DELETE
+Supported
+
+#### 3.3.3.4 SUBSCRIBE
+Supported
```

Comment:
Please add a detail support on telemetry like ON-CHANGE, SAMPLE etc.  You may also need to cover UT for telemetry.

Reply:
Done